pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -526,6 +571,26 @@ void connectionOpened(final ClientCnx cnx) {
     }
 
     /**
+     * Clear the internal receiver queue and returns the message id of what was the 1st message in the queue that was
+     * not seen by the application
+     */
+    private MessageIdImpl clearReceiverQueue() {
+        List<Message> currentMessageQueue = new ArrayList<>(incomingMessages.size());
+        incomingMessages.drainTo(currentMessageQueue);
+        if (!currentMessageQueue.isEmpty()) {
+            return (MessageIdImpl) currentMessageQueue.get(0).getMessageId();
+        } else if (lastDequeuedMessage != null) {
```

Comment:
Actually, this code is wrong... I had changed the logic for `startMessageId` from referring to the "first message to be returned" to the "last message that has been consumed"... but that wasn't reflected here.